docs/basic-features/data-fetching/getStaticPaths.md

```diff
@@ -0,0 +1,60 @@
+---
+description: Fetch data at build time with `getStaticProps` API reference.
+---
+
+# `getStaticPaths`
+
+If a page has [dynamic routes](/docs/routing/dynamic-routes.md) and uses `getStaticProps`, it needs to define a list of paths that have to be rendered to HTML at build time.
+
+When you export an `async` function called `getStaticPaths` (static generation) from a page that uses dynamic routes, Next.js will statically pre-render all the paths specified by `getStaticPaths`.
+
+```jsx
+export async function getStaticPaths() {
+  return {
+    paths: [
+      { params: { ... } }
+    ],
+    fallback: true // false or blocking
+  };
+}
+```
+
+The [`getStaticPaths` API reference](/docs/api-reference/data-fetching/getStaticPaths.md) covers all parameters and props that can be used with `getStaticPaths`.
+
+## When should I use `getStaticPaths`?
+
+You should use `getStaticPaths` if you’re statically pre-rendering pages that use dynamic routes.
+
+## TypeScript: Use `GetStaticPaths`
+
+For TypeScript, you can use the `GetStaticPaths` type from `next`:
+
+```ts
+import { GetStaticPaths } from 'next'
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  // ...
+}
+```
+
+## Technical details
+
+### Use together with [`getStaticProps`](/docs/basic-features/data-fetching/getStaticProps.md)
+
+When you use `getStaticProps` on a page with dynamic route parameters, you **must** use `getStaticPaths`.
+
+Note that you **cannot** use `getStaticPaths` with [`getServerSideProps`](/docs/basic-features/data-fetching/getServerSideProps.md).
+
+### Only runs at build time on server-side
```

Comment:
Using ISR, it can also run at request time (in the background). It _does_ only run on the server, though.